### PR TITLE
Fix #7021 : [UI bug] Extra rating factors overflow in the vehicle tab in Korean

### LINKED ISSUE #7021: [UI bug] Extra rating factors overflow in the vehicle tab in Korean
**OS:** Windows 10
**Version:** 0.1.2
**Commit/Build:** 27b57f8

Extra rating factors overflows in the vehicle tab in Korean as you can see in the below image.
No overflows in Chinese and Japanese.
I think it is about to line height.

**Screenshots / Video:**

image

**Save game:**

### PR DESCRIPTION
**Pull request by pss9205**
Change X,Y coordinates about Excitement,intensity,nausea factor
to fix the Bug which text is shown at outside of ui box.
Before
34909470-14afc202-f8e5-11e7-910a-3ae2f060fdd9
After
fixed1fixed2

### CODE REVIEW

File: contributors.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Contributors to OpenRCT2
+﻿# Contributors to OpenRCT2
```
Comment (janisozaur) [inline contributors.md]: Please remove this change

File: src/openrct2-ui/windows/Ride.cpp [hunk 2]
```diff
@@ -2915,33 +2915,33 @@ static void window_ride_vehicle_paint(rct_window *w, rct_drawpixelinfo *dpi)
...
     // Capacity
     gfx_draw_string_left(dpi, STR_CAPACITY, &rideEntry->capacity, COLOUR_BLACK, x, y);
-    y += 5;
 
     // Excitement Factor
     factor = rideEntry->excitement_multiplier;
     if (factor > 0)
     {
-        y += 10;
+        y += 12;
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/Ride.cpp]: Please use the `LIST_ROW_HEIGHT` constant.

File: src/openrct2-ui/windows/Ride.cpp [hunk 3]
```diff
@@ -2915,33 +2915,33 @@ static void window_ride_vehicle_paint(rct_window *w, rct_drawpixelinfo *dpi)
...
         gfx_draw_string_left(dpi, STR_EXCITEMENT_FACTOR, &factor, COLOUR_BLACK, x, y);
     }
 
     // Intensity Factor
     factor = rideEntry->intensity_multiplier;
     if (factor > 0)
     {
-        y += 10;
+		x += 150;
```
Comment (janisozaur) [inline src/openrct2-ui/windows/Ride.cpp]: Indentation should use spaces, not tabs

Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/Ride.cpp]: Could you apply the x offset in the call to `gfx_draw_string_left`? That way, you don't have to subtract it later on.

Also, and this is very minor, please indent using spaces, not tabs.

File: src/openrct2-ui/windows/Ride.cpp [hunk 4]
```diff
@@ -2915,33 +2915,33 @@ static void window_ride_vehicle_paint(rct_window *w, rct_drawpixelinfo *dpi)
...
         gfx_draw_string_left(dpi, STR_INTENSITY_FACTOR, &factor, COLOUR_BLACK, x, y);
     }
 
     // Nausea Factor
     factor = rideEntry->nausea_multiplier;
     if (factor > 0)
     {
-        y += 10;
+		x -= 150;
+		y += 12;
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/Ride.cpp]: See earlier comments.

File: contributors.md (hunk 1 above)
Comment (AaronVanGeffen) [inline contributors.md]: You've accidentally introduced a byte-ordering mark to this file. Please revert this change.

File: src/openrct2-ui/windows/Ride.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Ride.cpp]: Alternatively, you could if/else on `font_get_line_height(FONT_SPRITE_BASE_SMALL) == 10` and only apply the change for larger, non-sprite fonts.

**Comment by pss9205** [comment]
thanks for all of your comments. i changed things you mentioned.

**Comment by sjoerddebruin** [comment]
I agree with @AaronVanGeffen, can you test the change with (for example) German? One of the languages that are known for being longer than others.

**Comment by pss9205** [comment]
ok. accoding to your opinion, i added if/else to check the font size. now arabic, chinese, czech, japanese, korean, russian are affected by this change 
@AaronVanGeffen gfx_draw_string_left use 'FONT_SPRITE_BASE_MEDIUM'. so i checked the height of 'FONT_SPRITE_BASE_MEDIUM'. if i am wrong, i will change again
ENG-KOR
fixed1fixed2

**Comment by AaronVanGeffen** [comment]
@pss9205 I appreciate you changing it. There's a little too much code duplication going on now, though. I'll take care of it.

**Comment by AaronVanGeffen** [comment]
I've amended the code to remove the duplicated parts. The code now looks as follows.

German:
screenshot_20180123_144413

Korean:
screenshot_20180123_144427

As far as I'm concerned, this is ready to be (squash-)merged.

File: src/openrct2-ui/windows/Ride.cpp [hunk 5]
```diff
@@ -2919,54 +2919,36 @@ static void window_ride_vehicle_paint(rct_window *w, rct_drawpixelinfo *dpi)
...
-        // Excitement Factor
-        factor = rideEntry->excitement_multiplier;
-        if (factor > 0)
-        {
+
+    // Intensity Factor
+    factor = rideEntry->intensity_multiplier;
+    if (factor > 0)
+    {
+        if (font_get_line_height(FONT_SPRITE_BASE_MEDIUM) != 10)
```
Comment (rwjuk) [inline src/openrct2-ui/windows/Ride.cpp]: I'd store the result of `font_get_line_height(FONT_SPRITE_BASE_MEDIUM)` in a variable so you don't have to call the function twice.

File: src/openrct2-ui/windows/Ride.cpp [hunk 6]
```diff
@@ -2919,54 +2919,36 @@ static void window_ride_vehicle_paint(rct_window *w, rct_drawpixelinfo *dpi)
...
-        if (factor > 0)
-        {
-            y += 10;
-            gfx_draw_string_left(dpi, STR_NAUSEA_FACTOR, &factor, COLOUR_BLACK, x, y);
-        }
+    // Excitement Factor
+    factor = rideEntry->excitement_multiplier;
+    if (factor > 0)
+    {
+        y += LIST_ROW_HEIGHT;
```
Comment (pss9205) [inline src/openrct2-ui/windows/Ride.cpp]: @AaronVanGeffen  thanks for amending. but could you check for italian? italian has no extra margin in ui range. so if you add 12 to y, it will be out of ui range. my code add 10 to all y values in case font_get_line_height(FONT_SPRITE_BASE_MEDIUM) == 10
please check for this

Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Ride.cpp]: Hmm, I had tested with several languages, just not Italian. You're right, though — it is using four lines instead of two or three. Well, we'll have to tackle that at a later point — I had just merged this PR when you were typing your comment.

I suspect a further change won't be necessary, though, as the string is probably going to change soon. (This is already apparent when you compare the English version its the American English counterpart, for example.)
